image_get/worker/worker/util.py

```diff
@@ -0,0 +1,73 @@
+import asyncio
+import json
+import logging as log
+import time
+import pykafka
+import worker.settings as settings
+from functools import partial
+from io import BytesIO
+from PIL import Image, UnidentifiedImageError
+
+
+def kafka_connect():
+    try:
+        client = pykafka.KafkaClient(hosts=settings.KAFKA_HOSTS)
+    except pykafka.exceptions.NoBrokersAvailableError:
+        log.info('Retrying Kafka connection. . .')
+        time.sleep(3)
+        return kafka_connect()
+    return client
+
+
+def parse_message(message):
+    decoded = json.loads(str(message.value, 'utf-8'))
+    return decoded
+
+
+def thumbnail_image(img: Image):
+    img.thumbnail(size=settings.TARGET_RESOLUTION, resample=Image.NEAREST)
+    output = BytesIO()
+    img.save(output, format="JPEG", quality=30)
```

Comment:
I picked that number by decreasing the quality as much as I could without significant degradation of the image in a side-by-side comparison. Here's a comparison of rekognition performance on the original image vs our compressed thumbnail:

Full resolution image (not a thumbnail) with 100 quality:
![Full res with 100% quality](https://i.imgur.com/fEpFR95.png)

640x480 thumbnail with 30 quality:
![640x480 with 30% quality](https://i.imgur.com/isQbnZk.png)

I don't think Rekognition performance is impacted by this level of compression. Here's the [demo](https://console.aws.amazon.com/rekognition/home?region=us-east-1#/label-detection).

Reply:
I also did some tests with pics of my dog.  I get significant loss of confidence in the results.  How many pics did you test on?

Reply:
We might want to tune that with a 'random sample' of images from Flickr or something.

Reply:
Full size:
![image](https://user-images.githubusercontent.com/6829750/77672673-77d6f400-6f89-11ea-9560-810e31ecffc8.png)

Thumbnailed (modified for quality=80):
![image](https://user-images.githubusercontent.com/6829750/77673220-2e3ad900-6f8a-11ea-8c8b-cd5375471749.png)


Thumbnailed (modified for quality=70):
![image](https://user-images.githubusercontent.com/6829750/77673068-f764c300-6f89-11ea-92d8-0701a5c4d2d4.png)

Thumbnailed (modified for quality=60):
![image](https://user-images.githubusercontent.com/6829750/77673536-97225100-6f8a-11ea-90b7-a3166fcedcfc.png)


Thumbnailed (modified for quality=30):
![image](https://user-images.githubusercontent.com/6829750/77673436-735f0b00-6f8a-11ea-8087-8b069e2f0d34.png)

So, for that image at least, the 'labeling' goes away at quality=60

Reply:
We're not using the bounding boxes or the confidence intervals; we just need labels. In the first thumbnail example, with quality=80, the top tags are "strap, hound, mammal, dog, canine, animal, pet, outdoors". In the quality=30 example, the tags are "strap, clothing, apparel, dog, hound, canine, pet, animal, mammal". While they're not identical, they are all equally accurate (technically, since your dog is wearing a harness). I would need to see some other examples to be persuaded that 30 is too low.

You have convinced me that we need to do more testing though. Since this parameter is easily tunable, I'd like to defer these experiments to later, when the other project requirements are met. I'll add a followup ticket.

Reply:
Opened https://github.com/creativecommons/cccatalog-api/issues/471

Reply:
Totally agree we should defer the experiments.